### PR TITLE
Fix #1251 #1252 #1253

### LINKED ISSUE #1251: FortifyFpr: SourceBasePath from run on Linux causes exception
SourceBasePath will look like "/root/blah/blah" which causes a format exception to be thrown by the Uri constructor. We should add file: to the path string if it's from Linux.

### PR DESCRIPTION
**Pull request by EasyRhinoMSFT**
Fixes for issues uncovered by Katrina's Fortify FPR converter usage.

### CODE REVIEW

File: src/Sarif.Converters/FortifyFprConverter.cs [hunk 1]
```diff
@@ -124,11 +124,19 @@ public override void Convert(Stream input, IResultLogWriter output, OptionallyEm
...
             if (!string.IsNullOrWhiteSpace(_originalUriBasePath))
             {
-                var uri = new Uri(_originalUriBasePath);
-                run.OriginalUriBaseIds = new Dictionary<string, FileLocation>
+                if (_originalUriBasePath.StartsWith("/") &&
+                    _invocation.GetProperty("Platform") == "Linux")
                 {
-                    { FileLocationUriBaseId, new FileLocation { Uri = uri } }
-                };
+                    _originalUriBasePath = "file:/" + _originalUriBasePath;
```
Comment (lgolding) [inline src/Sarif.Converters/FortifyFprConverter.cs]: >/ [](start = 49, length = 1)

Reading RFC 3986, I think this should be just `file:`, not `file:/`.

File: src/Sarif.Converters.UnitTests/TestData/FortifyFprConverter/ExpectedOutputs/OneResultBasic.sarif [hunk 2]
```diff
@@ -11,7 +11,7 @@
       },
       "originalUriBaseIds": {
         "SRCROOT": {
-          "uri": "file:///C:/projects/myproject/src/"
+          "uri": "file://root/projects/myproject/src/"
```
Comment (lgolding) [inline src/Sarif.Converters.UnitTests/TestData/FortifyFprConverter/ExpectedOutputs/OneResultBasic.sarif]: >/ [](start = 23, length = 1)

Again, doesn't `file:/root...` work? If not, don't bother to change it.

**Comment by lgolding** [comment]
Before you provide a private package to Katrina, you need to change the version number to `...2019-01-09` in four places:

- build.props
- VersionConstants.cs
- sarif-schema.json
    - title
    - description

**Comment by lgolding** [comment]
Include that as part of this PR.

---
In reply to: [459772298](https://github.com/Microsoft/sarif-sdk/pull/1254#issuecomment-459772298) [](ancestors = 459772298)

File: src/Sarif.Converters/FortifyFprConverter.cs (hunk 1 above)
Comment (EasyRhinoMSFT) [reply src/Sarif.Converters/FortifyFprConverter.cs]: Tried that, Uri ctor throws.

---
In reply to: [253101090](https://github.com/Microsoft/sarif-sdk/pull/1254#discussion_r253101090) [](ancestors = 253101090)

File: src/Sarif.Converters.UnitTests/TestData/FortifyFprConverter/ExpectedOutputs/OneResultBasic.sarif (hunk 2 above)
Comment (EasyRhinoMSFT) [reply src/Sarif.Converters.UnitTests/TestData/FortifyFprConverter/ExpectedOutputs/OneResultBasic.sarif]: Meaning keep it as a relative URI?

---
In reply to: [253102144](https://github.com/Microsoft/sarif-sdk/pull/1254#discussion_r253102144) [](ancestors = 253102144)

Comment (EasyRhinoMSFT) [reply src/Sarif.Converters.UnitTests/TestData/FortifyFprConverter/ExpectedOutputs/OneResultBasic.sarif]: I'm going to check in with this change, we can revert later after we discuss.

---
In reply to: [253125417](https://github.com/Microsoft/sarif-sdk/pull/1254#discussion_r253125417) [](ancestors = 253125417,253102144)

**Pull request merged**